packages/controllers/src/utils.ts

```diff
@@ -52,6 +52,71 @@ export function isValidJson(value: unknown): value is Json {
   }
 }
 
+export function objectDiff<A, B>(a: A, b: B): Diff<A, B> {
+  return Object.entries(a).reduce((acc, [key, value]) => {
+    if (key in b) {
+      return acc;
+    }
+    acc[key] = value;
+    return acc;
+  }, {} as any);
```

Comment:
Is the cast needed here? Or can we use `reduce` generically?

Reply:
The cast is necessary since it's `{}` is not castable to `Diff<A,B>` as it lacks parameters of A.

Reply:
Hmm. @Mrtenz is there a way to type this reduce proper or would you also cast?

Reply:
@ritave @FrederikBolding I think in this case it might be better to do something like this:
```ts
return Object.entries(objectA).reduce<Record<string, unknown>>(
  (acc, [key, value]) => {
    if (!(key in objectB)) {
      acc[key] = value;
    }
    return acc;
  },
  {},
) as Diff<ObjectA, ObjectB>;
```

This keeps the reducer strictly typed, and only the result of that is casted to `Diff`.